public/docs/advanced/commands.md

```diff
@@ -1,129 +1,215 @@
 
-See the [Using the console](/docs/getting_started/console) section for more details.
 
-### compile
+### create
 
-Intelligently compile your contracts. This will only compile contracts that have changed since the last compile, unless otherwise specified.
+Helper to create new contracts, migrations and tests.
 
-```none
-$ truffle compile
+```shell
+truffle create (contract|migration|test) <ArtifactName>
 ```
 
-Optional parameter:
+Parameters:
 
-* `--all`: Compile all contracts instead of intelligently choosing.
-* `--network name`: Specify the network to use, saving artifacts specific to that network.
+* `contract`: Create a new contract definition and file `contracts/ArtifactName.sol`.
+* `migration`: Create a new migration and file `migrations/###########_artifact_name.js`.
+* `test`: Create a new test and file `tests/artifact_name.js`.
+* `<ArtifactName>`: Name of new artifact.
 
-### create contract
+Camel case names of artifacts will be converted to underscore-separated file names for the migrations and tests. Number prefixes for migrations are automatically generated.
 
-Helper method to scaffold a new contract. Name must be camel-case.
 
-```none
-$ truffle create contract MyContract
-```
+### debug
 
-### create migration
+*This feature is considered experimental.*
 
-Helper method to scaffold a new migration. Name must be camel-case.
+Interactively debug any transaction on the blockchain.
 
-```none
-$ truffle create migration MyContract
+```shell
+truffle debug <transaction_hash>
 ```
 
-### create test
+Will start an interactive debugging session on a particular transaction. Allows you to step through each action and replay. See the [Debugging your contracts](/docs/getting_started/debugging) section for more details.
+
+
+Parameters:
+
+* `<transaction_hash>`: Transaction ID to use for debugging.
+
+
+### deploy
+
+Alias for `migrate`. See `migrate` for details.
+
+
+### develop
 
-Helper method to scaffold a new test for a contract. Name must be camel-case.
+Open a console with a local TestRPC
 
-```none
-$ truffle create test MyTest
+```shell
+truffle develop
 ```
 
+Spawns a local development blockchain, and allows you to interact with contracts via the command line. Additionally, many Truffle commands are additionally available within the console (without the `truffle` prefix).
+
+If you want an interactive console but want to use an existing blockchain, use `truffle console`.
+
+See the [Using the console](/docs/getting_started/console) section for more details.
+
+
 ### exec
 
-Execute a Javascript file within the Truffle environment. This will include `web3`, set the default provider based on the network specified (if any), and include your contracts as global objects while executing the script. Your script must export a function that Truffle can run. See the [Writing external scripts](/docs/getting_started/scripts) section for more details.
+Execute a JS module within this Truffle environment.
```

Comment:
To me, `this` here is confusing. Which environment is dependent on the network specified (if specified). I'd just us `a`

Reply:
Not sure where I got "this" from. I like how it originally read now.